lang/cs/Org.Apache.REEF.Client/YARN/YARNREEFClient.cs

```diff
@@ -98,13 +99,50 @@ public IJobSubmissionResult SubmitAndGetJobStatus(JobRequest jobRequest)
         {
             var application = await _yarnClient.GetApplicationAsync(appId);
 
-            var msg = string.Format("application status {0}, Progress: {1}, trackingUri: {2}, Name: {3}, ApplicationId: {4}, State {5}.",
-                application.FinalStatus, application.Progress, application.TrackingUI, application.Name, application.Id, application.State);
-            Logger.Log(Level.Verbose, msg);
+            Logger.Log(Level.Verbose,
+               "application status {0}, Progress: {1}, uri: {2}, Name: {3}, ApplicationId: {4}, State {5}.",
+               application.FinalStatus,
+               application.Progress,
+               application.TrackingUI,
+               application.Name,
+               application.Id,
+               application.State);
 
             return application.FinalStatus;
         }
 
+        /// <summary>
+        /// Returns all the application reports running in the cluster.
+        /// GetApplicationReports call is very expensive as it is trying 
+        /// fetch information about all the applications in the cluster.
+        /// 
+        /// If this method is called right after submitting a new app then
+        /// that new app might not immediately result in this list until 
+        /// some number of retries. 
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IList<IApplicationReport>> GetApplicationReports()
```

Comment:
I think IDictionary<string, IApplication> will be better, with the key being applicationId. Doesn't seem like another implementation is necessary. The client can use whatever information he wants from the returned object. Also, if we modify the Applications class to return IList<IApplication> then we can just return applications.App.ToDictionary()